react-app/src/components/templates/MovieList/MovieListWrapperComponent.tsx

```diff
@@ -1,14 +1,21 @@
-import React, { FunctionComponent, useEffect, useState } from 'react';
-import { useSelector, useDispatch } from "react-redux";
+import React, { FunctionComponent, useEffect, Suspense } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { useParams } from 'react-router-dom';
 
-import { IMovie } from '../../molecules/MovieCard/IMovie';
 import { IMenuItem } from '../../molecules/Navigation/INavigation';
-import NavigationComponent from '../../molecules/Navigation/NavigationComponent';
-import MovieListComponent from '../../organisms/MovieList/MovieListComponent';
-import { MainWrapper, MovieCountWrapper } from './MovieListWrappers';
+import { MainWrapper, MovieCountWrapper, NoMovieFound } from './MovieListWrappers';
 import LoaderComponent from '../../atoms/Loader/LoaderComponent';
-import { actionApiFilterRequest } from '../../../actions/Actions';
+import {
+  actionApiFilterRequest,
+  actionSetFilter,
+  actionSelectMovieId,
+  actionControlVisibility,
+} from '../../../actions/Actions';
+import NavigationComponent from '../../molecules/Navigation/NavigationComponent';
 
+const HeaderComponent = React.lazy(() => import('../../organisms/Header/HeaderComponent'));
```

Comment:
You still seem to be using SSR, and `React.lazy` is [not (yet) supposed to work](https://reactjs.org/docs/code-splitting.html#reactlazy) in server-side mode. So, I'm not entirely sure how did you manage to get it working.